jenkins/docker-ecr/Jenkinsfile

```diff
@@ -0,0 +1,152 @@
+
+                    }
+                }
+            }
+        }
+        stage('docker-build-dockerhub') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    install_gcrane()
+                    git url: "$DOCKER_BUILD_GIT_REPOSITORY", branch: "$DOCKER_BUILD_GIT_REPOSITORY_REFERENCE"
+                    if (env.ENVIRONMENT == 'production') {
+                        def CREDENTIAL_ID = "jenkins-staging-docker-prod-token"
+                        sh "echo Switch to Production"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+                                gcrane cp opensearchstaging:${SOURCE_IMAGE_TAG_IN_STAGING} opensearchproject:${DESTINATION_IMAGE_TAG_IN_PROD}
+                            '''
+                        }
+                    } else {
+                        def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            '''
+                        }
+                    }
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                        sh "docker logout"
+                    }
+                }
+            }
+        }
+        stage('docker-build-ecr') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root'
+                }
+            }
+            steps {
+                script {
+                    install_gcrane()
+                    if (env.ENVIRONMENT == 'production')
```

Comment:
In ECR what benefits do we have for using more than one account?  `production` vs `staging` in our docker images is not much of a distinction as from an infrastructure perspective both are production systems for considerations of data loss or up-time.

Reply:
The docker images we release currently are released as staging and production. We should have this distinction in ECR as well.

We should not have data loss anywhere, production and staging. However, I believe production failure is much more critical than staging.

Reply:
Note; We discussed this offline, the names of these environments staging and production are not uniformly applied which creates some confusing.  Rather than have boolean's like 'is_production' lets accept paths and credential references to allow  for these to be placed in a deterministic way

Reply:
Following up on our discussion today, I have created an issue to support production promotion of artifacts via the promotion job